### DISCUSSION TITLE
Inference Processor

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#13670] [Unsupported Versions]
Hello!!

I'm looking to create some custom ingest pipelines using the inference processor. When using the dev tools to pull back all available processors I do not see the type "inference". Is there a manual way to add the inference processor somehow?

Currently running SO2.4.100 in standalone mode.

`GET _nodes/ingest?filter_path=nodes.*.ingest.processors`

`{
  "nodes": {
    "t7c89OznRLCF9fzmg5oCTw": {
      "ingest": {
        "processors": [
          {
            "type": "append"
          },
          {
            "type": "attachment"
          },
          {
            "type": "bytes"
          },
          {
            "type": "circle"
          },
          {
            "type": "community_id"
          },
          {
            "type": "convert"
          },
          {
            "type": "csv"
          },
          {
            "type": "date"
          },
          {
            "type": "date_index_name"
          },
          {
            "type": "dissect"
          },
          {
            "type": "dot_expander"
          },
          {
            "type": "drop"
          },
          {
            "type": "enrich"
          },
          {
            "type": "fail"
          },
          {
            "type": "fingerprint"
          },
          {
            "type": "foreach"
          },
          {
            "type": "geo_grid"
          },
          {
            "type": "geoip"
          },
          {
            "type": "grok"
          },
          {
            "type": "gsub"
          },
          {
            "type": "html_strip"
          },
          {
            "type": "join"
          },
          {
            "type": "json"
          },
          {
            "type": "kv"
          },
          {
            "type": "lowercase"
          },
          {
            "type": "network_direction"
          },
          {
            "type": "pipeline"
          },
          {
            "type": "redact"
          },
          {
            "type": "registered_domain"
          },
          {
            "type": "remove"
          },
          {
            "type": "rename"
          },
          {
            "type": "reroute"
          },
          {
            "type": "script"
          },
          {
            "type": "set"
          },
          {
            "type": "set_security_user"
          },
          {
            "type": "sort"
          },
          {
            "type": "split"
          },
          {
            "type": "trim"
          },
          {
            "type": "uppercase"
          },
          {
            "type": "uri_parts"
          },
          {
            "type": "urldecode"
          },
          {
            "type": "user_agent"
          }
        ]
      }
    }
  }
}`

**Comment by reyesj2** [comment]
something like this? 
``` 
{
      "inference": {
        "model_id": "foo",
        "target_field": "bar",
        "field_map": {
          "baz": "qux"
        },
        "inference_config": {
          "quux": "quuz"
        }
      }
    }
```

I got this from Kibana -> Stack Managment -> Ingest pipelines 
then creating a new pipeline and adding the inference processor. There is a button at the bottom right for "show request" 
![image](https://github.com/user-attachments/assets/585932c8-f52f-4dc1-885f-4a801f621d7b)

**Reply by ColeVan** [reply]
So the inference processor isn't available in the current 2.4.100 version using elastic 8.14.3. This is because xpack isn't set to true in the elasticsearch.yml configuration. After modifying it in the soc configuration tab the inference processor becomes available. The main reason I needed it was because I wanted to use AI Assistant and ESLER with the elastic knowledge base. Using all the features of security onion combined with backend elastic enterprise security gives our analysts the flexibility when hunting.
